plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestRetryingConnectionFactory.java

```diff
@@ -52,94 +55,56 @@ public void testEverythingImplemented()
         assertAllMethodsOverridden(ConnectionFactory.class, RetryingConnectionFactory.class);
     }
 
-    @Test
-    public void testSimplyReturnConnection()
-            throws Exception
-    {
-        Injector injector = getInjectorForActions(RETURN);
-        ConnectionFactory factory = injector.getInstance(RetryingConnectionFactory.class);
-        MockConnectorFactory mock = getMockFromInjector(injector);
-
-        Connection connection = factory.openConnection(SESSION);
-
-        assertThat(connection).isNotNull();
-        assertThat(mock.getCallCount()).isEqualTo(1);
-    }
-
-    @Test
-    public void testRetryAndStopOnTrinoException()
-    {
-        Injector injector = getInjectorForActions(THROW_SQL_RECOVERABLE_EXCEPTION, THROW_TRINO_EXCEPTION);
-        ConnectionFactory factory = injector.getInstance(RetryingConnectionFactory.class);
-        MockConnectorFactory mock = getMockFromInjector(injector);
-
-        assertThatThrownBy(() -> factory.openConnection(SESSION))
-                .isInstanceOf(TrinoException.class)
-                .hasMessage("Testing Trino exception");
-
-        assertThat(mock.getCallCount()).isEqualTo(2);
-    }
-
-    @Test
-    public void testRetryAndStopOnSqlException()
+    public static Object[][] happyPath()
     {
-        Injector injector = getInjectorForActions(THROW_SQL_RECOVERABLE_EXCEPTION, THROW_SQL_EXCEPTION);
-        ConnectionFactory factory = injector.getInstance(RetryingConnectionFactory.class);
-        MockConnectorFactory mock = getMockFromInjector(injector);
-
-        assertThatThrownBy(() -> factory.openConnection(SESSION))
-                .isInstanceOf(SQLException.class)
-                .hasMessage("Testing sql exception");
-
-        assertThat(mock.getCallCount()).isEqualTo(2);
+        return new Object[][] {{new Action[] {RETURN}, 1},
+                {new Action[] {THROW_SQL_RECOVERABLE_EXCEPTION, RETURN}, 2},
+                {new Action[] {THROW_WRAPPED_SQL_RECOVERABLE_EXCEPTION, RETURN}, 2}
+        };
     }
 
-    @Test
-    public void testNullPointerException()
+    public static Object[][] exceptionalPath()
     {
-        Injector injector = getInjectorForActions(THROW_NPE);
-        ConnectionFactory factory = injector.getInstance(RetryingConnectionFactory.class);
-        MockConnectorFactory mock = getMockFromInjector(injector);
-
-        assertThatThrownBy(() -> factory.openConnection(SESSION))
-                .isInstanceOf(NullPointerException.class)
-                .hasMessage("Testing NPE");
-
-        assertThat(mock.getCallCount()).isEqualTo(1);
+        return new Object[][] {{new Action[] {THROW_SQL_RECOVERABLE_EXCEPTION, THROW_TRINO_EXCEPTION}, 2, TrinoException.class, "Testing Trino exception"},
+                {new Action[] {THROW_SQL_RECOVERABLE_EXCEPTION, THROW_SQL_EXCEPTION }, 2, SQLException.class, "Testing sql exception"},
+                {new Action[] {THROW_NPE}, 1, NullPointerException.class, "Testing NPE"}
+        };
```

Comment:
This is the last commit as a 'icing on the top'. If you don't like this refactor, we can drop it.

My thinking was that if we ever add more retries or more exceptions, it's easier to edit a single line rather than to copy paste the whole block again and also change in all the methods for instance the 'isEqualTo(2)'.

I don't have a strong preference here. If for some reason you think that your proposal is more maintainable, I'm fine with doing that change.

Reply:
as said elsewhere the goal is to optimize for reading instead of writing as mentioned on https://trino.io/development/vision.html#development-philosophy

Reply:
What you are quoting could be interpreted one way or the other. I proposed the parametrized test refactor because it's more readable for my taste.

Anyway, in order to keep you satisfied dropped commit: dce25648de9399aca1937f4136b48ad3f7b35912 from this PR.